Rebase on 1.6.2

dbnicholson opened the pull request:
Rebase on 1.6.2 for the recent apps purchases work. Here's a list of what was done with the rebased commits:

pick dde9c0c3 Checkout using the same mode of the repo
pick f552b2c5 build-finish: also export speedwagon files
pick 570b7e7d build-finish: Add Endless Discovery Feed Providers
pick cba08af3 Allow defining an external location for the system installation and use it
pick 1f9a84a6 Include the external installation directory in XDG_DATA_DIRS
drop f1bc979c flatpak-dir: skip rewriting search provider .ini files for Endless apps (superseded by dir: don't disable search providers from com.endlessm)
pick 7a1f026a common/dir: Change default of core.add-remotes-config-dir to false
pick efc3f1ec profile/flatpak.sh: add export/bin dirs to PATH
pick 51eacfa1 Add Transifex translation infrastructure
drop b5fbd557 common: Do not convert G_IO_ERROR_CANCELLED errors to FLATPAK_ERROR domain when fetching remote metadata. (superseded by 142a7fd)
drop 910aa42b Use Python 3 for tests (upstream)
pick 7c59a871 common/dir: Skip parental controls checks when running from the eos-image-builder
pick f1f1e5f0 system-helper: Unconditionally allow 'app-install' polkit action
pick 5bab2805 Don't ignore .pot file
drop f05b5c72 Commit .pot file (all translations reapplied at end)
drop 53dfa8b6 Update .po files (all translations reapplied at end)
drop 6d2539e7 Reapply downstream translations (all translations reapplied at end)
drop 855f640c Updates from Transifex (all translations reapplied at end)
drop 66d012f8 utils: Allow chaining OstreeAsyncProgress when pushing GMainContext (upstream with some minor differences. need to make sure ostree version checks still apply correctly)
drop 484abd84 Revert "dir: Check commit signatures before resolving a ref" (upstream)
drop 9e908a2f Revert "flatpak-dir: skip rewriting search provider .ini files for Endless apps" (superseded by dir: don't disable search providers from com.endlessm)
pick a08e58a2 dir: don't disable search providers from com.endlessm
drop febc2a89 New translations from Transifex (all translations reapplied at end)
drop 68300f97 dir: Return empty array instead of NULL while querying related-refs (upstream)
drop 667754bb p2p: Don't mirror ostree-metadata refs when pulling into the child repo (upstream)

There are 4 more commits on the end. 3 of them are for translations, but "utils: Enable progress chaining on OSTree 2019.3" is needed to ensure the progress chaining still works with our ostree.

A diff just against 1.6.2 can be seen at https://github.com/endlessm/flatpak/compare/1.6.2...T29218-1.6.1-rebase.

A completed build is in https://obs-master.endlessm-sf.com/package/show/home:dbn:branches:eos:master:endless/flatpak

https://phabricator.endlessm.com/T29218

dbnicholson commented:
Debian packaging in #207

uajain commented:
I feel pretty good about what's here except for b5fbd55. I couldn't tell if 142a7fd should replace it entirely or not. @uajain?

Yes, 142a7fd should replace it b5fbd55 entirely, if I understand the rational given by Alex [here](https://github.com/flatpak/flatpak/pull/2932#issuecomment-566955077) correctly. I will also keep an eye if app-center starts to spew error-banners of "operation cancelled" when this rebase lands.

dbnicholson commented:
I've rebased this on 1.6.2 now and updated the notes in the first comment. I kept the same branch so the previous PR comments wouldn't be lost, although I don't see them now. The only real change is that I dropped b5fbd55 as @uajain advised.

dbnicholson commented:
Debian rebase in #207.

mwleeds commented:
Nice, the downstream patches we have are fairly minimal. Glad we don't have a gnome-software situation here.

mwleeds commented:
Review checklist (WIP):

- [x] The upstream patches since this release do not look urgently needed
- [ ] The downstream patches currently on master are all accounted for (either included or no longer needed)
- [x] The rebase is on the correct released commit
- [ ] The remaining downstream commits look correct
- [x] The CI is passing

mwleeds commented on system-helper/org.freedesktop.Flatpak.rules.in:
```diff
@@ -1,12 +1,14 @@
 polkit.addRule(function(action, subject) {
-    if ((action.id == "org.freedesktop.Flatpak.app-install" ||
-         action.id == "org.freedesktop.Flatpak.runtime-install"||
+    if ((action.id == "org.freedesktop.Flatpak.runtime-install"||
```
I realize this change is not new with this rebase, but wouldn't it make sense to also allow runtime installs? If the goal is for non-admins to be able to install apps.

dbnicholson replied:
Maybe, but I don't know the history of it. I just reapplied the change. The original discussion is in https://phabricator.endlessm.com/T27854#757523 and https://github.com/endlessm/flatpak/pull/185 and https://github.com/endlessm/flatpak/pull/187. I'd prefer to keep this as is and open a separate discussion for it if the behavior isn't right.

mwleeds replied:
Never mind, parental controls are only checked  for apps not runtimes, so that's the reason we special case `app-install`.

dbnicholson commented:
> This LGTM. I suggest that [2ef37d1](https://github.com/endlessm/flatpak/commit/2ef37d14b397b6ecdfddac2d63fa8020681499d4) could be upstreamed, though. We’re not carrying anything custom there, just what looks like some fixes to some incorrect translations. That could definitely be done as a follow-up though.

Yeah, @wjt suggested the same and then we potentially drop our flatpak transifex setup since the primary reason it was there was to get parental controls strings in ASAP. I filed https://phabricator.endlessm.com/T29427.
